src/backend/executor/functions.c

```diff
@@ -1278,8 +1278,6 @@ PG_TRY();
 		gp_enable_gpperfmon = false;
 	}
 
-	gp_enable_gpperfmon = orig_gp_enable_gpperfmon;
```

Comment:
Good catch for conflict resolution error from upstream. BUT! `fmgr_sql` function is about `language sql` functions. Did you test your solution against original reproduction steps with `language plpgsql`? Can you explain how plpgsql function call gets here?

Reply:
Yes, `fmgr_sql` handles only regular sql functions. `Plpgsql` and `do` statements are handled by `plpgsql_exec_function`, I saved `gp_enable_gpperfmon` at the beginning of that function and restored it at the end which enabled correct logging for such functions and its contents